### DISCUSSION TITLE
get-pnptenantesite - API permissions required

### DISCUSSION THREAD

**Discussion by temporaldoom** [pnp/powershell#4438] [Q&A]
I'm trying to give limited access to sharepoint via powershell in a read only scope. I wish for some users to run reporting scripts or be able to query Sharepoint data but not write to it.

I have granted an App Reg Sharepoint->Sites.Read.All and am authenticating to Sharepoint using a cert, using get-pnptenantsite returns attempted to perform an unauthorized function.

I would have hoped that SItes.Read.All would have the necessary permissions but it appears not to, unless I'm doing something terribly wrong with my AppReg/Authentication.

**Comment by enricobeyer** [comment]
Hello temporaldom,
do you have more details about the given permissions and the detailed script you used to get the information?

Best regards,
Enrico

**Comment by temporaldoom** [comment]
I have the Sites.Read.All permission sharepoint on the app reg, there is no detailed script. I used connect-pnponline with the appreg and then just used get-pnptenantsite.

**Comment by enricobeyer** [comment]
Hello temporaldom,

I guess only full control will work. It requires having SharePoint Online administrator access.

Best regards,
Enrico
